### ISSUE TITLE
More install errors

### ISSUE THREAD

**Issue by markab** [zfsonlinux/pkg-zfs#5]
Hi,

I just setup a completely fresh install of ubuntu 11.04 server but got the following errors, seems to be a problem with a spl devel package, but what do i need to install to fix this?

Thanks,
Mark

root@pluto:/etc# apt-get install debootstrap ubuntu-zfs
Reading package lists... Done
Building dependency tree  
Reading state information... Done
The following extra packages will be installed:
  binutils cpp cpp-4.5 dkms fakeroot gcc gcc-4.5 libavl0 libc-dev-bin libc6-dev libcloog-ppl0 libefi0 libelfg0 libgmpxx4ldbl libgomp1 libmpc2 libmpfr4 libnvpair0 libppl-c2 libppl7 libspl0
  libunicode0 libuutil0 libzfs0 libzpool0 linux-libc-dev make manpages-dev spl spl-dkms zfs-dkms zfsutils
Suggested packages:
  binutils-doc cpp-doc gcc-4.5-locales gcc-multilib autoconf automake1.9 libtool flex bison gdb gcc-doc gcc-4.5-multilib libmudflap0-4.5-dev gcc-4.5-doc libgcc1-dbg libgomp1-dbg libmudflap0-dbg
  binutils-gold glibc-doc make-doc zfs-initramfs
The following NEW packages will be installed
  binutils cpp cpp-4.5 debootstrap dkms fakeroot gcc gcc-4.5 libavl0 libc-dev-bin libc6-dev libcloog-ppl0 libefi0 libelfg0 libgmpxx4ldbl libgomp1 libmpc2 libmpfr4 libnvpair0 libppl-c2 libppl7
  libspl0 libunicode0 libuutil0 libzfs0 libzpool0 linux-libc-dev make manpages-dev spl spl-dkms ubuntu-zfs zfs-dkms zfsutils
0 upgraded, 34 newly installed, 0 to remove and 0 not upgraded.
Need to get 25.0 MB of archives.
After this operation, 78.6 MB of additional disk space will be used.
Do you want to continue [Y/n]? y
Get:1 http://gb.archive.ubuntu.com/ubuntu/ natty/main libgomp1 amd64 4.5.2-8ubuntu4 [25.4 kB]
Get:2 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main dkms all 2.1.1.2-5ubuntu1.1~natty [71.9 kB]
Get:3 http://gb.archive.ubuntu.com/ubuntu/ natty/main libgmpxx4ldbl amd64 2:4.3.2+dfsg-1ubuntu3 [10.4 kB]
Get:4 http://gb.archive.ubuntu.com/ubuntu/ natty/main libppl7 amd64 0.10.2-10ubuntu1 [357 kB]
Get:5 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main spl-dkms all 0.6.0.21-0ubuntu1~natty1 [430 kB]
Get:6 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main zfs-dkms amd64 0.6.0.21-0ubuntu1~natty1 [1,814 kB]
Get:7 http://gb.archive.ubuntu.com/ubuntu/ natty/main libppl-c2 amd64 0.10.2-10ubuntu1 [1,203 kB]
Get:8 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main spl amd64 0.6.0.21-0ubuntu1~natty1 [17.3 kB]
Get:9 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main libavl0 amd64 0.6.0.21-0ubuntu1~natty1 [19.9 kB]
Get:10 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main libefi0 amd64 0.6.0.21-0ubuntu1~natty1 [25.4 kB]
Get:11 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main libnvpair0 amd64 0.6.0.21-0ubuntu1~natty1 [38.3 kB]
Get:12 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main libspl0 amd64 0.6.0.21-0ubuntu1~natty1 [23.8 kB]
Get:13 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main libunicode0 amd64 0.6.0.21-0ubuntu1~natty1 [57.6 kB]
Get:14 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main libuutil0 amd64 0.6.0.21-0ubuntu1~natty1 [31.2 kB]
Get:15 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main libzpool0 amd64 0.6.0.21-0ubuntu1~natty1 [366 kB]
Get:16 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main libzfs0 amd64 0.6.0.21-0ubuntu1~natty1 [115 kB]
Get:17 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main zfsutils amd64 0.6.0.21-0ubuntu1~natty1 [271 kB]  
Get:18 http://ppa.launchpad.net/dajhorn/zfs/ubuntu/ natty/main ubuntu-zfs amd64 4~natty [1,678 B]  
Get:19 http://gb.archive.ubuntu.com/ubuntu/ natty/main libcloog-ppl0 amd64 0.15.9-2 [62.0 kB]  
Get:20 http://gb.archive.ubuntu.com/ubuntu/ natty/main libelfg0 amd64 0.8.13-1 [57.8 kB]  
Get:21 http://gb.archive.ubuntu.com/ubuntu/ natty/main libmpfr4 amd64 3.0.0-7 [188 kB]  
Get:22 http://gb.archive.ubuntu.com/ubuntu/ natty/main libmpc2 amd64 0.9-1 [39.7 kB]  
Get:23 http://gb.archive.ubuntu.com/ubuntu/ natty/main cpp-4.5 amd64 4.5.2-8ubuntu4 [4,603 kB]  
Get:24 http://gb.archive.ubuntu.com/ubuntu/ natty/main cpp amd64 4:4.5.2-1ubuntu3 [28.0 kB]  
Get:25 http://gb.archive.ubuntu.com/ubuntu/ natty-updates/main binutils amd64 2.21.0.20110327-2ubuntu3 [2,524 kB]  
Get:26 http://gb.archive.ubuntu.com/ubuntu/ natty/main gcc-4.5 amd64 4.5.2-8ubuntu4 [7,203 kB]  
Get:27 http://gb.archive.ubuntu.com/ubuntu/ natty/main gcc amd64 4:4.5.2-1ubuntu3 [5,062 B]  
Get:28 http://gb.archive.ubuntu.com/ubuntu/ natty/main make amd64 3.81-8ubuntu1 [119 kB]  
Get:29 http://gb.archive.ubuntu.com/ubuntu/ natty/main fakeroot amd64 1.14.4-1ubuntu1 [101 kB]  
Get:30 http://gb.archive.ubuntu.com/ubuntu/ natty/main libc-dev-bin amd64 2.13-0ubuntu13 [89.3 kB]  
Get:31 http://gb.archive.ubuntu.com/ubuntu/ natty/main linux-libc-dev amd64 2.6.38-8.42 [792 kB]  
Get:32 http://gb.archive.ubuntu.com/ubuntu/ natty/main libc6-dev amd64 2.13-0ubuntu13 [2,630 kB]  
Get:33 http://gb.archive.ubuntu.com/ubuntu/ natty/main manpages-dev all 3.27-1ubuntu2 [1,627 kB]  
Get:34 http://gb.archive.ubuntu.com/ubuntu/ natty/main debootstrap all 1.0.29ubuntu1 [35.2 kB]  
Fetched 25.0 MB in 38s (651 kB/s)  
Extract templates from packages: 100%
Selecting previously deselected package libgomp1.
(Reading database ... 48981 files and directories currently installed.)
Unpacking libgomp1 (from .../libgomp1_4.5.2-8ubuntu4_amd64.deb) ...
Selecting previously deselected package libgmpxx4ldbl.
Unpacking libgmpxx4ldbl (from .../libgmpxx4ldbl_2%3a4.3.2+dfsg-1ubuntu3_amd64.deb) ...
Selecting previously deselected package libppl7.
Unpacking libppl7 (from .../libppl7_0.10.2-10ubuntu1_amd64.deb) ...
Selecting previously deselected package libppl-c2.
Unpacking libppl-c2 (from .../libppl-c2_0.10.2-10ubuntu1_amd64.deb) ...
Selecting previously deselected package libcloog-ppl0.
Unpacking libcloog-ppl0 (from .../libcloog-ppl0_0.15.9-2_amd64.deb) ...
Selecting previously deselected package libelfg0.
Unpacking libelfg0 (from .../libelfg0_0.8.13-1_amd64.deb) ...
Selecting previously deselected package libmpfr4.
Unpacking libmpfr4 (from .../libmpfr4_3.0.0-7_amd64.deb) ...
Selecting previously deselected package libmpc2.
Unpacking libmpc2 (from .../libmpc2_0.9-1_amd64.deb) ...
Selecting previously deselected package cpp-4.5.
Unpacking cpp-4.5 (from .../cpp-4.5_4.5.2-8ubuntu4_amd64.deb) ...
Selecting previously deselected package cpp.
Unpacking cpp (from .../cpp_4%3a4.5.2-1ubuntu3_amd64.deb) ...
Selecting previously deselected package binutils.
Unpacking binutils (from .../binutils_2.21.0.20110327-2ubuntu3_amd64.deb) ...
Selecting previously deselected package gcc-4.5.
Unpacking gcc-4.5 (from .../gcc-4.5_4.5.2-8ubuntu4_amd64.deb) ...
Selecting previously deselected package gcc.
Unpacking gcc (from .../gcc_4%3a4.5.2-1ubuntu3_amd64.deb) ...
Selecting previously deselected package make.
Unpacking make (from .../make_3.81-8ubuntu1_amd64.deb) ...
Selecting previously deselected package dkms.
Unpacking dkms (from .../dkms_2.1.1.2-5ubuntu1.1~natty_all.deb) ...
Selecting previously deselected package spl-dkms.
Unpacking spl-dkms (from .../spl-dkms_0.6.0.21-0ubuntu1~natty1_all.deb) ...
Processing triggers for man-db ...
Setting up libgmpxx4ldbl (2:4.3.2+dfsg-1ubuntu3) ...
Setting up libppl7 (0.10.2-10ubuntu1) ...
Setting up libppl-c2 (0.10.2-10ubuntu1) ...
Setting up libcloog-ppl0 (0.15.9-2) ...
Setting up libelfg0 (0.8.13-1) ...
Setting up libmpfr4 (3.0.0-7) ...
Setting up libmpc2 (0.9-1) ...
Setting up cpp-4.5 (4.5.2-8ubuntu4) ...
Setting up cpp (4:4.5.2-1ubuntu3) ...
Setting up binutils (2.21.0.20110327-2ubuntu3) ...
Setting up libgomp1 (4.5.2-8ubuntu4) ...
Setting up gcc-4.5 (4.5.2-8ubuntu4) ...
Setting up gcc (4:4.5.2-1ubuntu3) ...
Setting up make (3.81-8ubuntu1) ...
Setting up dkms (2.1.1.2-5ubuntu1.1~natty) ...
Setting up spl-dkms (0.6.0.21-0ubuntu1~natty1) ...
Loading new spl-0.6.0.21 DKMS files...
First Installation: checking all kernels...
Building only for 2.6.38-8-server
Building initial module for 2.6.38-8-server
configure: error: in `/var/lib/dkms/spl/0.6.0.21/build':
configure: error: C compiler cannot create executables
See`config.log' for more details.

Error! Bad return status for module build on kernel: 2.6.38-8-server (x86_64)
Consult the make.log in the build directory
/var/lib/dkms/spl/0.6.0.21/build/ for more information.
Processing triggers for libc-bin ...
ldconfig deferred processing now taking place
Selecting previously deselected package zfs-dkms.
(Reading database ... 49545 files and directories currently installed.)
Unpacking zfs-dkms (from .../zfs-dkms_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package fakeroot.
Unpacking fakeroot (from .../fakeroot_1.14.4-1ubuntu1_amd64.deb) ...
Selecting previously deselected package libc-dev-bin.
Unpacking libc-dev-bin (from .../libc-dev-bin_2.13-0ubuntu13_amd64.deb) ...
Selecting previously deselected package linux-libc-dev.
Unpacking linux-libc-dev (from .../linux-libc-dev_2.6.38-8.42_amd64.deb) ...
Selecting previously deselected package libc6-dev.
Unpacking libc6-dev (from .../libc6-dev_2.13-0ubuntu13_amd64.deb) ...
Selecting previously deselected package manpages-dev.
Unpacking manpages-dev (from .../manpages-dev_3.27-1ubuntu2_all.deb) ...
Selecting previously deselected package spl.
Unpacking spl (from .../spl_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package libavl0.
Unpacking libavl0 (from .../libavl0_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package libefi0.
Unpacking libefi0 (from .../libefi0_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package libnvpair0.
Unpacking libnvpair0 (from .../libnvpair0_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package libspl0.
Unpacking libspl0 (from .../libspl0_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package libunicode0.
Unpacking libunicode0 (from .../libunicode0_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package libuutil0.
Unpacking libuutil0 (from .../libuutil0_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package libzpool0.
Unpacking libzpool0 (from .../libzpool0_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package libzfs0.
Unpacking libzfs0 (from .../libzfs0_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package zfsutils.
Unpacking zfsutils (from .../zfsutils_0.6.0.21-0ubuntu1~natty1_amd64.deb) ...
Selecting previously deselected package ubuntu-zfs.
Unpacking ubuntu-zfs (from .../ubuntu-zfs_4~natty_amd64.deb) ...
Selecting previously deselected package debootstrap.
Unpacking debootstrap (from .../debootstrap_1.0.29ubuntu1_all.deb) ...
Processing triggers for man-db ...
Processing triggers for ureadahead ...
Setting up zfs-dkms (0.6.0.21-0ubuntu1~natty1) ...
Loading new zfs-0.6.0.21 DKMS files...
First Installation: checking all kernels...
Building only for 2.6.38-8-server
Building initial module for 2.6.38-8-server
configure: error:
    **\* Please make sure the spl devel package for your distribution
    **\* is installed then try again.  If that fails you can specify the
    **\* location of the spl source with the '--with-spl=PATH' option.

Error! Bad return status for module build on kernel: 2.6.38-8-server (x86_64)
Consult the make.log in the build directory
/var/lib/dkms/zfs/0.6.0.21/build/ for more information.
Setting up fakeroot (1.14.4-1ubuntu1) ...
update-alternatives: using /usr/bin/fakeroot-sysv to provide /usr/bin/fakeroot (fakeroot) in auto mode.
Setting up libc-dev-bin (2.13-0ubuntu13) ...
Setting up linux-libc-dev (2.6.38-8.42) ...
Setting up libc6-dev (2.13-0ubuntu13) ...
Setting up manpages-dev (3.27-1ubuntu2) ...
Setting up spl (0.6.0.21-0ubuntu1~natty1) ...
Setting up libavl0 (0.6.0.21-0ubuntu1~natty1) ...
Setting up libefi0 (0.6.0.21-0ubuntu1~natty1) ...
Setting up libnvpair0 (0.6.0.21-0ubuntu1~natty1) ...
Setting up libspl0 (0.6.0.21-0ubuntu1~natty1) ...
Setting up libunicode0 (0.6.0.21-0ubuntu1~natty1) ...
Setting up libuutil0 (0.6.0.21-0ubuntu1~natty1) ...
Setting up libzpool0 (0.6.0.21-0ubuntu1~natty1) ...
Setting up libzfs0 (0.6.0.21-0ubuntu1~natty1) ...
Setting up zfsutils (0.6.0.21-0ubuntu1~natty1) ...
Setting up ubuntu-zfs (4~natty) ...
Setting up debootstrap (1.0.29ubuntu1) ...
Processing triggers for libc-bin ...
ldconfig deferred processing now taking place

**Comment by markab** [comment]
Here is the config.log

root@pluto:/var/lib/dkms/spl/0.6.0.21/build# more config.log
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by configure, which was
generated by GNU Autoconf 2.63.  Invocation command line was

  $ /var/lib/dkms/spl/0.6.0.21/build/configure --prefix=/usr --with-config=kernel --with-linux=/lib/modules/2.6.38-8-server/build --with-linux-obj=/lib/modules/2.6.38-8-server/build

## ---------

## Platform.

## ---------

hostname = pluto
uname -m = x86_64
uname -r = 2.6.38-8-server
uname -s = Linux
uname -v = #42-Ubuntu SMP Mon Apr 11 03:49:04 UTC 2011

/usr/bin/uname -p = unknown
/bin/uname -X     = unknown

/bin/arch              = unknown
/usr/bin/arch -k       = unknown
/usr/convex/getsysinfo = unknown
/usr/bin/hostinfo      = unknown
/bin/machine           = unknown
/usr/bin/oslevel       = unknown
/bin/universe          = unknown

PATH: /usr/local/sbin
PATH: /usr/local/bin
PATH: /usr/sbin
PATH: /usr/bin
PATH: /sbin
PATH: /bin
PATH: /usr/games
PATH: /usr/sbin
PATH: /sbin
PATH: /usr/lib/dkms

## -----------

## Core tests.

## -----------

configure:2084: checking metadata
configure:2257: result: yes
configure:2298: checking build system type
configure:2316: result: x86_64-unknown-linux-gnu
configure:2338: checking host system type
configure:2353: result: x86_64-unknown-linux-gnu
configure:2375: checking target system type
configure:2390: result: x86_64-unknown-linux-gnu
configure:2419: checking whether to enable maintainer-specific portions of Makefiles
configure:2428: result: no
configure:2469: checking for a BSD-compatible install
configure:2537: result: /usr/bin/install -c
configure:2548: checking whether build environment is sane
configure:2608: result: yes
configure:2749: checking for a thread-safe mkdir -p
configure:2788: result: /bin/mkdir -p
configure:2801: checking for gawk
configure:2831: result: no
configure:2801: checking for mawk
configure:2817: found /usr/bin/mawk
configure:2828: result: mawk
configure:2839: checking whether make sets $(MAKE)
configure:2861: result: yes
configure:2960: checking for a BSD-compatible install
configure:3028: result: /usr/bin/install -c
configure:3087: checking for gcc
configure:3103: found /usr/bin/gcc
configure:3114: result: gcc
configure:3346: checking for C compiler version
configure:3354: gcc --version >&5
gcc (Ubuntu/Linaro 4.5.2-8ubuntu4) 4.5.2
Copyright (C) 2010 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

configure:3358: $? = 0
configure:3365: gcc -v >&5
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/x86_64-linux-gnu/gcc/x86_64-linux-gnu/4.5.2/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu/Linaro 4.5.2-8ubuntu4' --with-bugurl=file:///usr/share/doc/gcc-4.5/README.Bugs --enable-languages=c,c++,fortran,objc,obj-c++ --prefix=
/usr --program-suffix=-4.5 --enable-shared --enable-multiarch --with-multiarch-defaults=x86_64-linux-gnu --enable-linker-build-id --with-system-zlib --libexecdir=/usr/lib/x86_64-linux-gnu --without
-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.5 --libdir=/usr/lib/x86_64-linux-gnu --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug -
-enable-libstdcxx-time=yes --enable-plugin --enable-gold --enable-ld=default --with-plugin-ld=ld.gold --enable-objc-gc --disable-werror --with-arch-32=i686 --with-tune=generic --enable-checking=rel
ease --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.5.2 (Ubuntu/Linaro 4.5.2-8ubuntu4) 
configure:3369: $? = 0
configure:3376: gcc -V >&5
gcc: '-V' option must have argument
configure:3380: $? = 1
configure:3403: checking for C compiler default output file name
configure:3425: gcc    conftest.c  >&5
/usr/bin/ld: cannot find crt1.o: No such file or directory
/usr/bin/ld: cannot find crti.o: No such file or directory
/usr/bin/ld: cannot find -lc
/usr/bin/ld: cannot find crtn.o: No such file or directory
collect2: ld returned 1 exit status
configure:3429: $? = 1
configure:3467: result: 
configure: failed program was:
| /\* confdefs.h.  _/
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define SPL_META_NAME "spl"
| #define SPL_META_VERSION "0.6.0.21"
| #define SPL_META_RELEASE "rc4"
| #define SPL_META_ALIAS "spl-0.6.0.21-rc4"
| #define PACKAGE "spl"
| #define VERSION "0.6.0.21"
| /_ end confdefs.h.  */
| 
| int
| main ()
| {
| 
|   ;
|   return 0;
| }
configure:3473: error: in `/var/lib/dkms/spl/0.6.0.21/build':
configure:3476: error: C compiler cannot create executables
See`config.log' for more details.

## ----------------

## Cache variables.

## ----------------

ac_cv_build=x86_64-unknown-linux-gnu
ac_cv_env_CC_set=
ac_cv_env_CC_value=
ac_cv_env_CFLAGS_set=
ac_cv_env_CFLAGS_value=
ac_cv_env_CPPFLAGS_set=
ac_cv_env_CPPFLAGS_value=
ac_cv_env_CPP_set=
ac_cv_env_CPP_value=
ac_cv_env_LDFLAGS_set=
ac_cv_env_LDFLAGS_value=
ac_cv_env_LIBS_set=
ac_cv_env_LIBS_value=
ac_cv_env_build_alias_set=
ac_cv_env_build_alias_value=
ac_cv_env_host_alias_set=
ac_cv_env_host_alias_value=
ac_cv_env_target_alias_set=
ac_cv_env_target_alias_value=
ac_cv_host=x86_64-unknown-linux-gnu
ac_cv_path_install='/usr/bin/install -c'
ac_cv_path_mkdir=/bin/mkdir
ac_cv_prog_AWK=mawk
ac_cv_prog_ac_ct_CC=gcc
ac_cv_prog_make_make_set=yes
ac_cv_target=x86_64-unknown-linux-gnu

## -----------------

## Output variables.

## -----------------

ACLOCAL='${SHELL} /var/lib/dkms/spl/0.6.0.21/build/config/missing --run aclocal-1.11'
ALIEN=''
ALIEN_VERSION=''
AMDEPBACKSLASH=''
AMDEP_FALSE=''
AMDEP_TRUE=''
AMTAR='${SHELL} /var/lib/dkms/spl/0.6.0.21/build/config/missing --run tar'
AM_BACKSLASH='\'
AM_DEFAULT_VERBOSITY='1'
AR=''
AUTOCONF='${SHELL} /var/lib/dkms/spl/0.6.0.21/build/config/missing --run autoconf'
AUTOHEADER='${SHELL} /var/lib/dkms/spl/0.6.0.21/build/config/missing --run autoheader'
AUTOMAKE='${SHELL} /var/lib/dkms/spl/0.6.0.21/build/config/missing --run automake-1.11'
AWK='mawk'
CC='gcc'
CCDEPMODE=''
CFLAGS=''
CONFIG_KERNEL_FALSE=''
CONFIG_KERNEL_TRUE=''
CONFIG_USER_FALSE=''
CONFIG_USER_TRUE=''
CPP=''
CPPFLAGS=''
CYGPATH_W='echo'
DEBUG_CFLAGS=''
DEFAULT_PACKAGE=''
DEFS=''
DEPDIR=''
DPKG=''
DPKGBUILD=''
DPKGBUILD_VERSION=''
DPKG_VERSION=''
DSYMUTIL=''
DUMPBIN=''
ECHO_C=''
ECHO_N='-n'
ECHO_T=''
EGREP=''
EXEEXT=''
FGREP=''
GREP=''
HAVE_ALIEN=''
HAVE_DPKG=''
HAVE_DPKGBUILD=''
HAVE_RPM=''
HAVE_RPMBUILD=''
INSTALL_DATA='${INSTALL} -m 644'
INSTALL_PROGRAM='${INSTALL}'
INSTALL_SCRIPT='${INSTALL}'
INSTALL_STRIP_PROGRAM='$(install_sh) -c -s'
KERNELCPPFLAGS=''
KERNELMAKE_PARAMS=''
LD=''
LDFLAGS=''
LIBOBJS=''
LIBS=''
LIBTOOL=''
LICENSE=''
LINUX=''
LINUX_OBJ=''
LINUX_SYMBOLS=''
LINUX_VERSION=''
LIPO=''
LN_S=''
LTLIBOBJS=''
MAINT='#'
MAINTAINER_MODE_FALSE=''
MAINTAINER_MODE_TRUE='#'
MAKEINFO='${SHELL} /var/lib/dkms/spl/0.6.0.21/build/config/missing --run makeinfo'
MKDIR_P='/bin/mkdir -p'
NM=''
NMEDIT=''
OBJDUMP=''
OBJEXT=''
OTOOL64=''
OTOOL=''
PACKAGE='spl'
PACKAGE_BUGREPORT=''
PACKAGE_NAME=''
PACKAGE_STRING=''
PACKAGE_TARNAME=''
PACKAGE_VERSION=''
PATH_SEPARATOR=':'
RANLIB=''
RPM=''
RPMBUILD=''
RPMBUILD_VERSION=''
RPM_VERSION=''
SED=''
SET_MAKE=''
SHELL='/bin/bash'
SPL_CONFIG=''
SPL_META_ALIAS='spl-0.6.0.21-rc4'
SPL_META_AUTHOR=''
SPL_META_DATA=''
SPL_META_LT_AGE=''
SPL_META_LT_CURRENT=''
SPL_META_LT_REVISION=''
SPL_META_NAME='spl'
SPL_META_RELEASE='rc4'
SPL_META_VERSION='0.6.0.21'
STRIP=''
VENDOR=''
VERSION='0.6.0.21'
ac_ct_CC='gcc'
ac_ct_DUMPBIN=''
am__EXEEXT_FALSE=''
am__EXEEXT_TRUE=''
am__fastdepCC_FALSE=''
am__fastdepCC_TRUE=''
am__include=''
am__isrc=''
am__leading_dot='.'
am__quote=''
am__tar='${AMTAR} chof - "$$tardir"'
am__untar='${AMTAR} xf -'
bindir='${exec_prefix}/bin'
build='x86_64-unknown-linux-gnu'
build_alias=''
build_cpu='x86_64'
build_os='linux-gnu'
build_vendor='unknown'
datadir='${datarootdir}'
datarootdir='${prefix}/share'
docdir='${datarootdir}/doc/${PACKAGE}'
dvidir='${docdir}'
exec_prefix='NONE'
host='x86_64-unknown-linux-gnu'
host_alias=''
host_cpu='x86_64'
host_os='linux-gnu'
host_vendor='unknown'
htmldir='${docdir}'
includedir='${prefix}/include'
infodir='${datarootdir}/info'
install_sh='${SHELL} /var/lib/dkms/spl/0.6.0.21/build/config/install-sh'
libdir='${exec_prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
lt_ECHO='echo'
mandir='${datarootdir}/man'
mkdir_p='/bin/mkdir -p'
oldincludedir='/usr/include'
pdfdir='${docdir}'
prefix='/usr'
program_transform_name='s,x,x,'
psdir='${docdir}'
sbindir='${exec_prefix}/sbin'
sharedstatedir='${prefix}/com'
sysconfdir='${prefix}/etc'
target='x86_64-unknown-linux-gnu'
target_alias=''
target_cpu='x86_64'
target_os='linux-gnu'
target_vendor='unknown'

## -----------

## confdefs.h.

## -----------

#define PACKAGE_NAME ""
#define PACKAGE_TARNAME ""
#define PACKAGE_VERSION ""
#define PACKAGE_STRING ""
#define PACKAGE_BUGREPORT ""
#define SPL_META_NAME "spl"
#define SPL_META_VERSION "0.6.0.21"
#define SPL_META_RELEASE "rc4"
#define SPL_META_ALIAS "spl-0.6.0.21-rc4"
#define PACKAGE "spl"
#define VERSION "0.6.0.21"

configure: exit 77
root@pluto:/var/lib/dkms/spl/0.6.0.21/build#

**Comment by dajhorn** [comment]
I can reproduce this error on a vanilla Ubuntu Natty server installation.  It is happening because the compiler is installed and invoked before the libc6-dev package is installed.  Doing this avoids the error:

```
 # apt-get install --no-install-recommends libc6-dev
 # apt-get install ubuntu-zfs
```

At first glance, this bug seems like it is caused by the "Pre-Depends: spl-dkms" control in the zfs-dkms package.

**Comment by markab** [comment]
Ok thanks

**Comment by Nowaker** [comment]
I just got the same error. Installing `libc6-dev` before `ubuntu-zfs zfs-initramfs` resolved the problem.

**Comment by dajhorn** [comment]
This issue seems resolved. Please reopen the ticket if the installation failure happens again.

**Comment by Rudd-O** [comment]
I have this problem again if I installed zfs-dkms without ubuntu-zfs.  This is a common thing to do and it is not clear that one must install ubuntu-zfs.  Please fix!

**Comment by Rudd-O** [comment]
Setting up ubuntu-zfs (6~lucid) ...
Processing triggers for libc-bin ...
ldconfig deferred processing now taking place
root@sfor3s37:/home/manuel# zfs list
Failed to load ZFS module stack.
Load the module manually by running 'insmod <location>/zfs.ko' as root.
Failed to load ZFS module stack.
Load the module manually by running 'insmod <location>/zfs.ko' as root.
root@sfor3s37:/home/manuel# dpkg-reconfigure spl-dkms

---

Deleting module version: 0.6.0.56

## completely from the DKMS tree.

Done.
Loading new spl-0.6.0.56 DKMS files...
Building only for 3.0.0-17-server
Building initial module for 3.0.0-17-server
configure: error: in `/var/lib/dkms/spl/0.6.0.56/build':
configure: error: C compiler cannot create executables
See`config.log' for more details.

**Comment by Rudd-O** [comment]
libc6-dev is missing as a dependency of dkms:

root@sfor3s37:/home/manuel# apt-get install libc6-dev
Reading package lists... Done
Building dependency tree  
Reading state information... Done
The following extra packages will be installed:
  libc-dev-bin linux-libc-dev
Suggested packages:
  glibc-doc
Recommended packages:
  manpages-dev
The following NEW packages will be installed:
  libc-dev-bin libc6-dev linux-libc-dev
0 upgraded, 3 newly installed, 0 to remove and 0 not upgraded.
Need to get 3,778kB of archives.
After this operation, 16.7MB of additional disk space will be used.
Get:1 http://apt-repo/ubuntu/ lucid-updates/main libc-dev-bin 2.11.1-0ubuntu7.10 [224kB]
Get:2 http://apt-repo/ubuntu/ lucid-updates/main linux-libc-dev 2.6.32-40.87 [836kB]
Get:3 http://apt-repo/ubuntu/ lucid-updates/main libc6-dev 2.11.1-0ubuntu7.10 [2,717kB]
Fetched 3,778kB in 2s (1,562kB/s)  
Selecting previously deselected package libc-dev-bin.
(Reading database ... 51988 files and directories currently installed.)
Unpacking libc-dev-bin (from .../libc-dev-bin_2.11.1-0ubuntu7.10_amd64.deb) ...
Selecting previously deselected package linux-libc-dev.
Unpacking linux-libc-dev (from .../linux-libc-dev_2.6.32-40.87_amd64.deb) ...
Selecting previously deselected package libc6-dev.
Unpacking libc6-dev (from .../libc6-dev_2.11.1-0ubuntu7.10_amd64.deb) ...
Processing triggers for man-db ...
Setting up libc-dev-bin (2.11.1-0ubuntu7.10) ...
Setting up linux-libc-dev (2.6.32-40.87) ...
Setting up libc6-dev (2.11.1-0ubuntu7.10) ...

workaround is to add it as a dependency in spl-dkms and zfs-dkms.

**Comment by Rudd-O** [comment]
Please reopen and fix!

**Comment by dajhorn** [comment]
Ubuntu specifically and intentionally allows this failure mode to accommodate certain advanced system configurations;  things like local `make-kpkg` installations and other kernel overrides.

This behavior is by design.  Implementing your recommendation would affect people that use this feature, especially those that do custom installations for cloud deployments.

Open a bug report against dkms if you still think that it should depend on libc6-dev or any other package.  If the upstream maintainer agrees, then I will kludge the bug in pkg-zfs until the fix is published in regular distribution.
